packages/core/src/Component.ts

```diff
@@ -0,0 +1,8 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+export interface Component {
```

Comment:
> Why are we doing this?

This doesn't work when the dep is not declared in the same file -- ie. from the Array of classes passed in. We can pair in a later session to refactor if you have other ideas that actually work.